python/paddle/fluid/layers/nn.py

```diff
@@ -5649,6 +5650,52 @@ def flatten(x, axis=1, name=None):
     return out
 
 
+def sequence_enumerate(input, win_size, pad_value, name=None):
+    """
+    Generate a new LoDTensor 
+    with the same 1st dimension length as the original LoDTensor, 
+    and with the 2nd dimension equal to the input window length, 
+    the new sub-sequence on 2nd dimension is enumerated one by one on the original sequence.
+    The values of the last insufficient part areall filled with the input pad_value.
+    
+    Examples:
+    Case 1:
+      Input:
+        X.lod = [[0, 3, 5]]
+        X.data = [[1], [2], [3], [4], [5]]
+        X.dims = [5, 1]
+      Attrs:
+        win_size = 2
+        pad_value = 0
+      Output:
+        Out.lod = [[0, 3, 5]]
+        Out.data = [[1, 2], [2, 3], [3, 0], [4, 5], [5, 0]]
+        Out.dims = [5, 2]
+
+    Args:
+        input (Variable): The input variable which is a LoDTensor
+        win_size (int): The enumerate sequence window size.
+        pad_value (int): The enumerate sequence padding value.
```

Comment:
`pad_value (int): The padding value, default 0.`

Reply:
Done